spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/core/GcpContextAutoConfiguration.java

```diff
@@ -72,4 +79,35 @@ public GcpProjectIdProvider gcpProjectIdProvider() {
 
 		return projectIdProvider;
 	}
+
+	/**
+	 * Provides default implementation for determining GCP environment.
+	 * Can be overridden to avoid interacting with real environment.
+	 * @since 1.1
+	 */
+	@Bean
+	@ConditionalOnMissingBean
+	public static GcpEnvironmentProvider gcpEnvironmentProvider() {
```

Comment:
In core refdoc, mention both EnvironmentProvider,  ResourceManager and MetadataProvider.